src/EFCore.PG/Query/Internal/NpgsqlQueryableMethodTranslatingExpressionVisitor.cs

```diff
@@ -0,0 +1,234 @@
+                            shapedQueryExpression = shapedQueryExpression.UpdateResultCardinality(ResultCardinality.Single);
+                            return CheckTranslated(
+                                TranslateStringAggregate(shapedQueryExpression, methodCallExpression.Arguments[1], selector: null));
+
+                        case nameof(NpgsqlQueryableExtensions.StringAggregate)
+                            when genericMethod == NpgsqlQueryableExtensions.StringAggregateWithSelectorMethod:
+                            shapedQueryExpression = shapedQueryExpression.UpdateResultCardinality(ResultCardinality.Single);
+                            return CheckTranslated(
+                                TranslateStringAggregate(
+                                    shapedQueryExpression,
+                                    methodCallExpression.Arguments[1],
+                                    selector: GetLambdaExpressionFromArgument(2)));
+
+                    }
+
+                    LambdaExpression GetLambdaExpressionFromArgument(int argumentIndex)
+                        => methodCallExpression.Arguments[argumentIndex].UnwrapLambdaFromQuote();
+                }
+            }
+
+            return base.VisitMethodCall(methodCallExpression);
+        }
+
+        // Copied from RelationalQueryableMethodTranslatingExpressionVisitor.TranslateMax
+        protected virtual ShapedQueryExpression TranslateStringAggregate(
+            ShapedQueryExpression source, Expression delimiter, LambdaExpression selector)
+        {
+            Check.NotNull(source, nameof(source));
+            Check.NotNull(delimiter, nameof(delimiter));
+
+            var delimiterTranslation = TranslateExpression(delimiter);
+            if (delimiterTranslation is null)
+            {
+                return null;
+            }
+
+            var selectExpression = (SelectExpression)source.QueryExpression;
+            selectExpression.PrepareForAggregate();
+            HandleGroupByForAggregate(selectExpression);
+
+            var newSelector = selector == null
+                              || selector.Body == selector.Parameters[0]
+                ? selectExpression.Projection.Count == 0
+                    ? selectExpression.GetMappedProjection(new ProjectionMember())
+                    : null
+                : RemapLambdaBody(source, selector);
+
+            if (newSelector == null)
+            {
+                return null;
+            }
+
+            var translatedSelector = TranslateExpression(newSelector);
+            if (translatedSelector == null)
+            {
+                return null;
+            }
+
+            var projection = _sqlTranslator.TranslateStringAggregate(translatedSelector, delimiterTranslation);
+
+            return AggregateResultShaper(source, projection, throwWhenEmpty: true, typeof(string));
+        }
+
+        #region Copied from RelationalQueryableMethodTranslatingExpressionVisitor
```

Comment:
Note code copied below, which seems like it would be necessary for anyone needing to add top-level aggregate function support. Ideally this duplication wouldn't be necessary.